xarray/core/alignment.py

```diff
@@ -502,19 +502,19 @@ def reindex_variables(
 
     Returns
     -------
-    reindexed : OrderedDict
+    reindexed : dict
         Dict of reindexed variables.
-    new_indexes : OrderedDict
+    new_indexes : dict
         Dict of indexes associated with the reindexed variables.
     """
     from .dataarray import DataArray
 
     # create variables for the new dataset
-    reindexed = OrderedDict()  # type: OrderedDict[Any, Variable]
+    reindexed = {}  # type: Dict[Any, Variable]
```

Comment:
This does work. But we don't seem to use this syntax elsewhere. For example: 

https://github.com/pydata/xarray/blob/4c05d38459dcf20ac4dc7fb32bb8cef4d7627c8b/xarray/core/dataset.py#L502

I feel like @crusaderky will have opinions here but for now, I'd like to leave the annotations with in their current format.

Reply:
This syntax is only Python 3.6+. So at this point might as well switch over type comments to variable annotations. But I agree, it isn't urgent.

Reply:
Yes it was just a suggestion. Merge away!

Reply:
In my PR that migrated from 3.5 to 3.6 I got rid of almost all the ``# type: ...`` annotations, but I couldn't remove the ``# type: OrderedDict[...]`` because typing.OrderedDict was added in 3.7.2